python/prairielearn.py

```diff
@@ -52,6 +53,45 @@ class QuestionData(TypedDict):
 class ElementTestData(QuestionData):
     test_type: Literal['correct', 'incorrect', 'invalid']
 
+
+def set_weighted_score_data(data: QuestionData, weight_default: int=1) -> None:
+    """
+    Sets main question score to be weighted average of all partial scores. Uses
+    weight_default to fill in a default weight for a question if one is missing.
```

Comment:
It looks like you're using "question" to refer to both "a question" and "a scored element within a question". This is a bit confusing and isn't used to refer to pieces of questions anywhere else, so I'd like to avoid introducing it.

```suggestion
    Sets overall question score to be weighted average of all partial scores. Uses
    weight_default to fill in a default weight for a score if one is missing.
```